allennlp/service/server_flask.py

```diff
@@ -162,39 +168,44 @@ def predict(model_name: str) -> Response:  # pylint: disable=unused-variable
 
         log_blob = {"model": model_name, "inputs": data, "cached": False, "outputs": {}}
 
-        # See if we hit or not. In theory this could result in false positives.
-        pre_hits = _caching_prediction.cache_info().hits  # pylint: disable=no-value-for-parameter
+        if cache_flag:
```

Comment:
I feel like you made this too complicated, I would have just changed line 169 to

```
if check_cache and cache_size > 0
```

and then line 182 to

```
if record_to_database and demo_db is not None:
```

is there a reason why that wouldn't just work?